src/nvim/charset.c

```diff
@@ -1392,6 +1392,22 @@ long getdigits_long(char_u **pp, bool strict, long def)
   return (long)number;
 }
 
+/// Gets a int32_t number from a string.
+///
+/// @see getdigits
+int32_t getdigits_int32(char **pp, bool strict, long def)
+{
+  intmax_t number = getdigits((char_u **)pp, strict, def);
+#if SIZEOF_INTMAX_T > SIZEOF_INT32_T
```

Comment:
```suggestion
#if SIZEOF_INTMAX_T > sizeof(int32_t)
```

Reply:
I don't understand why this has to be a macro though. Using

```c
  if (SIZEOF_INTMAX_T > sizeof(int32_t)) {
    ...
  }

```

works just as well.

Reply:
> I don't understand why this has to be a macro though.

To avoid coverity et al warning about useless checks.